maven-plugin-api/src/main/java/org/apache/maven/plugin/AbstractMojo.java

```diff
@@ -144,55 +144,39 @@
     implements Mojo, ContextEnabled
 {
     /** Instance logger */
-    private Log log;
+    private final Log log = LogFactory.getLog( getClass() );
```

Comment:
open question: makes the lookup earlier, not sure it is good in all cases + it now prevents to reset the log which was sometimes done with extension components

Reply:
back to the roots (as it was)